### PR TITLE
[Feature][Connector-V2][Http] Add option rules && Improve Myhours sink connector

### LINKED ISSUE #3350: [Feature][Connector V2] expose configurable options in Http @TyrantLucifer

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

Add option rules && Improve Myhours sink connector

close #3350 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
@Hisoka-X @EricJoy2048 @TaoZex @hailin0 PTAL.

File: seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/sink/HttpSinkFactory.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    @Override
+    public String factoryIdentifier() {
+        return "Http";
+    }
+
+    @Override
+    public OptionRule optionRule() {
+        return OptionRule.builder()
+                .required(HttpConfig.URL)
+                .optional(HttpConfig.HEADERS)
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/sink/HttpSinkFactory.java:38]: `optional` support multi-parameter.

Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/sink/HttpSinkFactory.java:38]: Would chain calls be more clearly? What do you think about it?

**Review by Hisoka-X — DISMISSED** [review]

File: seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/HttpParameter.java [hunk 2]
```diff
@@ -40,35 +40,35 @@ public class HttpParameter implements Serializable {
...
+        if (pluginConfig.hasPath(HttpConfig.POLL_INTERVAL_MILLS.key())) {
+            this.setPollIntervalMillis(pluginConfig.getInt(HttpConfig.POLL_INTERVAL_MILLS.key()));
         }
-        if (pluginConfig.hasPath(HttpConfig.RETRY)) {
-            this.setRetry(pluginConfig.getInt(HttpConfig.RETRY));
-            if (pluginConfig.hasPath(HttpConfig.RETRY_BACKOFF_MULTIPLIER_MS)) {
-                this.setRetryBackoffMultiplierMillis(pluginConfig.getInt(HttpConfig.RETRY_BACKOFF_MULTIPLIER_MS));
+        if (pluginConfig.hasPath(HttpConfig.RETRY.key())) {
+            this.setRetry(pluginConfig.getInt(HttpConfig.RETRY.key()));
+            if (pluginConfig.hasPath(HttpConfig.RETRY_BACKOFF_MULTIPLIER_MS.key())) {
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/HttpParameter.java:74]: Where do you use `RETRY_BACKOFF_MULTIPLIER_MS .defaultValue()` ?

Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/HttpParameter.java:74]: PTAL @hailin0

File: seatunnel-connectors-v2/connector-http/connector-http-myhours/src/main/java/org/apache/seatunnel/connectors/seatunnel/myhours/source/config/MyHoursSourceParameter.java [hunk 3]
```diff
@@ -29,36 +28,27 @@
...
-        //put authorization in headers
+        // put authorization in headers
         this.headers = this.getHeaders() == null ? new HashMap<>() : this.getHeaders();
-        this.headers.put(MyHoursSourceConfig.AUTHORIZATION, MyHoursSourceConfig.ACCESSTOKEN_PREFIX + " " + accessToken);
+        this.headers.put(MyHoursSourceConfig.AUTHORIZATION, MyHoursSourceConfig.ACCESS_TOKEN_PREFIX + " " + accessToken);
         this.setHeaders(this.headers);
     }
 
     public void buildWithLoginConfig(Config pluginConfig) {
+        super.buildWithConfig(pluginConfig);
```
Comment (TaoZex) [inline seatunnel-connectors-v2/connector-http/connector-http-myhours/src/main/java/org/apache/seatunnel/connectors/seatunnel/myhours/source/config/MyHoursSourceParameter.java]: How about writing set retry as a separate method? This line of code only uses the set retry part, and other connectors can be reused in the future.

File: seatunnel-connectors-v2/connector-http/connector-http-myhours/src/main/java/org/apache/seatunnel/connectors/seatunnel/myhours/source/MyHoursSource.java [hunk 4]
```diff
@@ -55,12 +54,13 @@ public String getPluginName() {
...
-        CheckResult result = CheckConfigUtil.checkAllExists(pluginConfig, MyHoursSourceConfig.URL, MyHoursSourceConfig.EMAIL, MyHoursSourceConfig.PASSWORD);
+        CheckResult result = CheckConfigUtil.checkAllExists(pluginConfig, MyHoursSourceConfig.URL.key(),
+                MyHoursSourceConfig.EMAIL.key(), MyHoursSourceConfig.PASSWORD.key());
         if (!result.isSuccess()) {
             throw new PrepareFailException(getPluginName(), PluginType.SOURCE, result.getMsg());
         }
         //Login to get accessToken
-        String accessToken = null;
+        String accessToken;
         accessToken = getAccessToken(pluginConfig);
```
Comment (TaoZex) [inline seatunnel-connectors-v2/connector-http/connector-http-myhours/src/main/java/org/apache/seatunnel/connectors/seatunnel/myhours/source/MyHoursSource.java]:
```suggestion
        String accessToken = getAccessToken(pluginConfig);
```

File: seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/HttpParameter.java (hunk 2 above)
Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/HttpParameter.java:74]: Fixed, PTAL

File: seatunnel-connectors-v2/connector-http/connector-http-myhours/src/main/java/org/apache/seatunnel/connectors/seatunnel/myhours/source/MyHoursSource.java (hunk 4 above)
Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-http/connector-http-myhours/src/main/java/org/apache/seatunnel/connectors/seatunnel/myhours/source/MyHoursSource.java]: Resolved, PTAL

File: seatunnel-connectors-v2/connector-http/connector-http-myhours/src/main/java/org/apache/seatunnel/connectors/seatunnel/myhours/source/config/MyHoursSourceParameter.java (hunk 3 above)
Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-http/connector-http-myhours/src/main/java/org/apache/seatunnel/connectors/seatunnel/myhours/source/config/MyHoursSourceParameter.java]: Resolved

**Review by TaoZex — APPROVED** [review]
LGTM

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**